integration/storagetest/logtests.go

```diff
@@ -465,15 +467,17 @@ func (*logTests) TestDequeueLeaves(ctx context.Context, t *testing.T, s storage.
 		t.Fatalf("Failed to queue leaves: %v", err)
 	}
 
-	{
-		// Now try to dequeue them
-		runLogTX(s, tree, t, func(ctx context.Context, tx2 storage.LogTreeTX) error {
+	// Now try to dequeue them
+	cctx, cancel := context.WithTimeout(ctx, 3*time.Second) // Retry until timeout
```

Comment:
Could you talk a bit about why we need this timeout?

Reply:
ReadWriteTransaction will retry `codes.Aborted` forever unless there is a timeout :-)

Reply:
aha, great - thanks, possibly worth popping this in a comment there for others who pass by later on?

Reply:
done